templates/environment/partials/nat-gateways.yml

```diff
@@ -0,0 +1,30 @@
+{{- range $ind, $cidr := .PrivateSubnetCIDRs}}
+NatGateway{{inc $ind}}Attachment:
+  Type: AWS::EC2::EIP
+  DependsOn: InternetGatewayAttachment
+  Properties:
+    Domain: vpc
+NatGateway{{inc $ind}}:
+  Type: AWS::EC2::NatGateway
+  Properties:
+    AllocationId: !GetAtt NatGateway{{inc $ind}}Attachment.AllocationId
+    SubnetId: !Ref PublicSubnet{{inc $ind}}
+    Tags:
+      - Key: Name
+        Value: !Sub 'copilot-${AppName}-${EnvironmentName}-{{$ind}}'
```

Comment:
It won't matter much but I am wondering if we should use `{{inc $ind}}` here? In case customers are looking at the created infrastructure and starts wondering why is natgateway2attachment attached natgateway1.

Reply:
I am 100% with you! I had the same thought, but for the sake of consistency with the numbering of subnet tag Names, I followed suit here. 

At least we're consistently inconsistent-- the CFN logical IDs (in this realm, at least) are numbered starting with 1, but the Name values in Tags start with 0. 
In the CFN template, users might see:
```
  NatGateway2:
    Type: AWS::EC2::NatGateway
    Properties:
      AllocationId: !GetAtt NatGateway2Attachment.AllocationId
      SubnetId: !Ref PublicSubnet2
```
And in the VPC console they'd see that subnet with tagged Name copilot-app-env-pub1 and logicalID PublicSubnet2.

It's definitely not ideal, but I'd be nervous about changing existing numbering.